UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/AzureBlobStorageClient.cs

```diff
@@ -0,0 +1,110 @@
+﻿using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+using Newtonsoft.Json;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using UKHO.ExchangeSetService.Common.Configuration;
+using UKHO.ExchangeSetService.Common.Logging;
+using UKHO.ExchangeSetService.Common.Models.SalesCatalogue;
+using UKHO.ExchangeSetService.Common.Storage;
+
+namespace UKHO.ExchangeSetService.Common.Helpers
+{
+    public class AzureBlobStorageClient : IAzureBlobStorageClient
+    {
+        private readonly IScsStorageService scsStorageService;
+        private const string CONTENT_TYPE = "application/json";
+        private readonly IOptions<EssFulfilmentStorageConfiguration> storageConfig;
+        private readonly IAzureMessageQueueHelper azureMessageQueueHelper;
+        private readonly ILogger<AzureBlobStorageClient> logger;
+        public AzureBlobStorageClient(IScsStorageService scsStorageService, IOptions<EssFulfilmentStorageConfiguration> storageConfig, IAzureMessageQueueHelper azureMessageQueueHelper, ILogger<AzureBlobStorageClient> logger)
+        {
+            this.scsStorageService = scsStorageService;
+            this.storageConfig = storageConfig;
+            this.azureMessageQueueHelper = azureMessageQueueHelper;
+            this.logger = logger;
+        }
+
+        public async Task<bool> StoreScsResponseAsync(string containerName, string batchId, SalesCatalogueResponse salesCatalogueResponse, CancellationToken cancellationToken)
+        {
+            string uploadFileName = string.Concat(batchId, ".json");
+
+            string storageAccountConnectionString =
+                  scsStorageService.GetStorageAccountConnectionString();
+            CloudBlockBlob cloudBlockBlob = GetCloudBlockBlob(uploadFileName, storageAccountConnectionString, containerName);
+            cloudBlockBlob.Properties.ContentType = CONTENT_TYPE;
+
+            await UploadScsResponseToBlobAsync(cloudBlockBlob, salesCatalogueResponse);
+
+            await AddQueueMessage(batchId, salesCatalogueResponse, cloudBlockBlob);
+
+            logger.LogInformation(EventIds.SCSResponseStoredAndSentMessageInQueue.ToEventId(), "Sales catalogue response saved for the {batchId}", batchId);
+            return true;
+        }
```

Comment:
This method could be named StoreSalesCatalogueStoreResponseAsync.

I realise this is a little longer but would make it clear to somebody who is new to the code exactly what its doing.

Reply:
Sorry StoreSaleCatalogueServiceResponseAsync

Reply:
Done